### PR TITLE
redo translucency options

### LINKED ISSUE #1494: Translucency issue
Setting "Enable Translucency" to No does not disable translucency.  This can be replicated by warping to map 30 in Doom 2 and looking at the megasphere.

### LINKED ISSUE #1508: Draw percentage signs for menu items if applicable
Somewhat related to bobbing percentages: a menu-item flag to be used with numeric/thermo items to make a percentage sign be drawn alongside the value.

Crude text mockup:
`Translucency [—————|] 100%`

This would be much more concise than having to write "percent(age)" in the item description.

I haven't implemented such a thing myself, and while it sounds simple enough, I'm not particularly familiar with the relevant code, so I think it'd be best if some of you implemented it -- if you like the idea, of course.

### PR DESCRIPTION
**Pull request by rfomin**
* Rename "Enable Translucency"->"Sprite Translucency"

* Instantly enable/disable translucency for things

* Make a thermo for the translucency percentage

Fix #1494 

Fix #1508 in 04b5c74

### CODE REVIEW

**Comment by fabiangreffrath** [comment]
Do things which have their translucency flag set by DEHACKED remain translucent regardless of that switch?

**Comment by rfomin** [comment]
> Do things which have their translucency flag set by DEHACKED remain translucent regardless of that switch?

Oh, now I remember why we added that code. I don't know any PWAD that defines `TRANSLUCENT` flag other than [Heretic Invades Doom](https://www.doomworld.com/forum/topic/123082-heretic-invades-doom-two-small-maps-for-mbf21-ports/), and it's MBF21 only. Maybe we should simplify this?

**Comment by fabiangreffrath** [comment]
> Oh, now I remember why we added that code. I don't know any PWAD that defines `TRANSLUCENT` flag other than [Heretic Invades Doom](https://www.doomworld.com/forum/topic/123082-heretic-invades-doom-two-small-maps-for-mbf21-ports/), and it's MBF21 only. Maybe we should simplify this?

I like that we let the PWAD author decide which sprites are translucent and which aren't.

**Comment by rfomin** [comment]
> I like that we let the PWAD author decide which sprites are translucent and which aren't.

I tried to implement that for [Heretic Invades Doom](https://www.doomworld.com/forum/topic/123082-heretic-invades-doom-two-small-maps-for-mbf21-ports/) (the only PWAD that uses `TRANSLUCENCY` flag that I know), and it's not easy. The disciples turn their transparency flag on and off during their attack (DEHACKED line 648) using MFB21 codepointer `A_AddFlags`. I think it's not practical to support this.

My suggestion is to simplify it for now. Such functionality is not in demand.

**Comment by fabiangreffrath** [comment]
> turn their transparency flag on and off during their attack

Okay, that's complicated. Let's go easy and straight then...

File: src/m_menu.c [hunk 1]
```diff
@@ -2412,6 +2412,8 @@ static void M_DrawSetting(setup_menu_t *s, int accum_y)
 
       if (strings)
         strcpy(menu_buffer, strings[value]);
+      else if (flags & S_THRM_PCT)
+        M_snprintf(menu_buffer, 4, "%d%%", value);
```
Comment (ceski-1) [inline src/m_menu.c:2416]: `%` is cut off for `100%`.

Comment (rfomin) [reply src/m_menu.c]: Fixed in https://github.com/fabiangreffrath/woof/pull/1496/commits/44f9e9822c427f2199d154d4fa6ec6fdcb152767

**Review by fabiangreffrath — APPROVED** [review]

**Pull request merged**